src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the state of the availability of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is presented, the block data is not available.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when a header is a part of the chain that can potentially replace our consensus tip 
+        /// because its tip's total chain work is greater comparing to what we currently have.
+        /// </summary>
+        BlockRequired,
+
+        /// <summary>
+        /// The <see cref="Block"/> was downloaded and is available, but it may not be reachable directly but via a store.
+        /// </summary>
+        BlockAvailable
+    }
+
+    /// <summary>
+    /// Represents the validation level of a block.
+    /// </summary>
+    public enum ValidationState
+    {
+        /// <summary>
+        /// No validation was performed.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// We have a valid block header.
+        /// </summary>
+        HeaderValidated,
+
+        /// <summary>
+        /// Minimum validation required because the block is assumed to be valid.
+        /// This state is used for blocks before the last checkpoint or for blocks that are on the chain of assume valid block.
+        /// Once a block is fully validated this state will change to FullyValidated.
+        /// </summary>
+        AssumedValid,
+
+        /// <summary>
+        /// Validated using all rules that don't require change of state.
```

Comment:
No I am avoiding specific tech, coinview is a name of a class representing UTXO state.

Reply:
I think in case when a component is well known and came from BTC core it's ok to use particular name such as a coinview. It is more understandable this way because `don't require change of state.` doesn't say much. Which state? What is a state and when can it be changed?

Reply:
i'd just add
`/// Some rules validation may be skipped for blocks marked as <see cref="AssumedValid"/>. `

and i'm ok with this